embedded-graphics/src/primitives/arc/plane_sector.rs

```diff
@@ -0,0 +1,150 @@
+
+        if self.sweep < ANGLE_180DEG {
+            correct_a_side && correct_b_side
+        } else if self.sweep < ANGLE_360DEG {
+            correct_a_side || correct_b_side
+        } else {
+            true
+        }
+    }
+}
+
+/// Iterator that returns only the points which are inside a plane sector defined by two lines.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub(in crate::primitives) struct PlaneSectorIterator {
+    plane_sector: PlaneSector,
+    points: rectangle::Points,
+}
+
+impl PlaneSectorIterator {
+    pub(in crate::primitives) fn new<D: Dimensions>(
+        primitive: &D,
+        center: Point,
+        angle_start: Angle,
+        angle_sweep: Angle,
+    ) -> Self {
+        Self {
+            plane_sector: PlaneSector::new(center, angle_start, angle_sweep),
+            points: primitive.bounding_box().points(),
+        }
+    }
+
+    pub(in crate::primitives) fn empty() -> Self {
+        Self {
+            plane_sector: PlaneSector::empty(),
+            points: Rectangle::new(Point::zero(), Size::zero()).points(),
+        }
+    }
+}
+
+impl Iterator for PlaneSectorIterator {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let plane_sector = self.plane_sector;
+        self.points.find(|p| plane_sector.contains(*p))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        geometry::AngleUnit,
+        primitives::{circle::DistanceIterator, Arc},
+    };
+
+    #[test]
+    fn plane_arc_iter() {
+        let arc = Arc::new(Point::zero(), 3, 0.0.deg(), 90.0.deg());
+
+        let mut iter =
+            PlaneSectorIterator::new(&arc, arc.center(), arc.angle_start, arc.angle_sweep);
+        assert_eq!(iter.next(), Some(Point::new(1, 0)));
+        assert_eq!(iter.next(), Some(Point::new(2, 0)));
+        assert_eq!(iter.next(), Some(Point::new(1, 1)));
+        assert_eq!(iter.next(), Some(Point::new(2, 1)));
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn plane_sector_iter_empty() {
+        let mut iter = PlaneSectorIterator::empty();
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn distance_iter() {
+        let arc = Arc::new(Point::zero(), 3, 0.0.deg(), 90.0.deg());
+
+        let mut iter = DistanceIterator::new(
+            arc.center_2x(),
+            PlaneSectorIterator::new(&arc, arc.center(), arc.angle_start, arc.angle_sweep),
+        );
+        assert_eq!(iter.next(), Some((Point::new(1, 0), 4)));
+        assert_eq!(iter.next(), Some((Point::new(2, 0), 8)));
+        assert_eq!(iter.next(), Some((Point::new(1, 1), 0)));
+        assert_eq!(iter.next(), Some((Point::new(2, 1), 4)));
+        assert_eq!(iter.next(), None);
+    }
+
+    #[test]
+    fn distance_iter_empty() {
+        let mut iter = DistanceIterator::new(Point::zero(), PlaneSectorIterator::empty());
+        assert_eq!(iter.next(), None);
+    }
```

Comment:
IMO we don't need these anymore, because the they test `DistanceIter`, which is in another module.

Reply:
Good point. I thought it would be useful to test the combination of `DistanceIterator` and `PlaneSectorIterator` but they're pretty redundant with the two tests before these. I'll remove them.